submissions/7SergSmith7/Friends-app/app.js

```diff
@@ -0,0 +1,143 @@
+import "../css/vendor/normalize.css";
+import "../css/styles.css";
+
+const USERS_URL =
+  "https://randomuser.me/api/?inc=gender,phone,picture,dob,location,name&results=25";
+
+const contactsField = document.querySelector(".contacts-field");
+const optionsMenu = document.querySelector(".options");
+const searchInput = document.querySelector(".options__search__inpit");
+let contactsData;
+let tempContacts;
+let filter = "";
+
+function getContacts(url) {
+  return fetch(url)
+    .then((res) => res.json())
+    .then(({ results }) => (contactsData = results))
+    .then((contactsData) => {
+      copyData();
+      renderContacts(contactsData);
+    });
+}
+
+function renderContactTemplate(contact) {
+  const { gender, name, location, dob, phone, picture } = contact;
+  return ` <div class="contact__item" data-name="${name.first}">
+    <img class="contact__image" src=${picture.large} alt=${name.first}>
+    <h3 class="contact__title">${name.first} ${name.last}</h3>
+    <p class="contact__phone">${phone}</p>   
+    <div class="contact__additional">
+    <div class="contact__additional__gender">
+    <h4 class="additional__gender__title">sex</h4>
+    <p class="additional__gender_item">${gender}</p>
+    </div>
+    <div class="contact__additional__age">
+    <h4 class="additional__age__title">age</h4>
+    <p class="additional__age_item">${dob.age}</p>
+    </div>
+    <div class="contact__additional__location">
+    <h4 class="additional__location__title">location</h4>   
+    <p class="additional__location_item">${location.country}</p>     
+    </div>
+    
+    </div>
+</div>
+`;
+}
+
+function renderContacts(contacts) {
+  contactsField.innerHTML = contacts
+    .map((contact) => renderContactTemplate(contact))
+    .join("");
+}
+optionsMenu.addEventListener("click", onOptionsClick);
+
+function onOptionsClick(e) {
+  const selectedOption = e.target.classList.value;
+
+  switch (selectedOption) {
+    case "name-sort__z-a":
+      sortName(tempContacts);
+      renderContacts(sortWithGender(tempContacts, filter));
+
+      break;
+    case "name-sort__a-z":
+      sortName(tempContacts);
+      tempContacts = tempContacts.reverse();
+      renderContacts(sortWithGender(tempContacts, filter));
+
+      break;
+    case "age-sort__more":
+      sortAge(tempContacts);
+
+      renderContacts(sortWithGender(tempContacts, filter));
+
+      break;
+    case "age-sort__less":
+      sortAge(tempContacts);
+      tempContacts = tempContacts.reverse();
+      renderContacts(sortWithGender(tempContacts, filter));
+
+      break;
+    case "gender-filter__man":
+      filter = "male";
+      renderContacts(filterGender(tempContacts, filter));
+
+      break;
+    case "gender-filter__woman":
+      filter = "female";
+      renderContacts(filterGender(tempContacts, filter));
```

Comment:
`sortWithGender` and `filterGender` - functions named by different conventions. Please, choose one. This will increase readability of the code.